pkg/apis/hive/v1/hiveconfig_types.go

```diff
@@ -194,6 +214,67 @@ type ManageDNSAzureConfig struct {
 	ResourceGroupName string `json:"resourceGroupName"`
 }
 
+// ControllerGoroutinesConfig contains number of goroutines for each controller
+type ControllerGoroutinesConfig struct {
+	// Default specifies default number of goroutines for each Controller
+	// +optional
+	Default *int `json:"default,omitempty"`
+	// ControllerConfigs is a list of goroutines configurations for each controller
+	// +optional
+	ControllerConfigs []ControllerConfig `json:"configs,omitempty"`
+}
+
+// ControllerClientQPSConfig contains client rate limiter QPS for each controller
+type ControllerClientQPSConfig struct {
+	// ControllerClientQPSConfig contains allowed client rate limiter QPS for each controller
+	// +optional
+	Default *int `json:"default,omitempty"`
+	// ControllerConfigs is a list of client rate limiter QPS configurations for each controller
+	// +optional
+	ControllerConfigs []ControllerConfig `json:"configs,omitempty"`
+}
+
+// ControllerClientBurstConfig contains client rate limiter burst for each controller
+type ControllerClientBurstConfig struct {
+	// Default specifies default client rate limiter burst for each controller
+	// +optional
+	Default *int `json:"default,omitempty"`
+	// ControllerConfigs is a list of client rate limiter burst configurations for each controller
+	// +optional
+	ControllerConfigs []ControllerConfig `json:"configs,omitempty"`
+}
+
+// ControllerQueueQPSConfig contains allowed workqueue rate limiter QPS for each controller
+type ControllerQueueQPSConfig struct {
+	// Default specifies default workqueue rate limiter QPS for each controller
+	// +optional
+	Default *int `json:"default,omitempty"`
+	// ControllerConfigs is a list of workqueue rate limiter QPS configurations for each controller
+	// +optional
+	ControllerConfigs []ControllerConfig `json:"configs,omitempty"`
+}
+
+// ControllerQueueBurstConfig contains workqueue rate limiter burst for each controller
+type ControllerQueueBurstConfig struct {
+	// Default specifies default workqueue rate limiter burst for each controller
+	// +optional
+	Default *int `json:"default,omitempty"`
+	// ControllerConfigs is a list of workqueue rate limiter burst configurations for each controller
+	// +optional
+	ControllerConfigs []ControllerConfig `json:"configs,omitempty"`
+}
+
+// ControllerConfig contains the configuration for a particular controller
+type ControllerConfig struct {
+	// name specifies the name of the controller
+	// +kubebuilder:validation:Enum=clusterdeployment;clusterrelocate;clusterstate;clusterversion;controlPlaneCerts;dnsendpoint;dnszone;remoteingress;remotemachineset;syncidentityprovider;syncset;syncsetinstance;unreachable;velerobackup;clusterprovision;clusterDeprovision;clusterpool;clusterpoolnamespace
+	// +optional
+	Name string `json:"name,omitempty"`
+	// value specifies the value of the configuration
+	// +optional
+	Value *int `json:"value,omitempty"`
```

Comment:
I think both name and value here should not be optional. If you're using this struct, both fields seem to be required.